core/trino-main/src/main/java/io/trino/sql/planner/optimizations/LimitPushDown.java

```diff
@@ -135,8 +135,10 @@ public PlanNode visitLimit(LimitNode node, RewriteContext<LimitContext> context)
             }
 
             if (!node.requiresPreSortedInputs() && (!node.isWithTies() || (limit != null && node.getCount() >= limit.getCount()))) {
+                // The new limit context is partial if neither the existing context nor this limit node is final
+                boolean partial = node.isPartial() && (limit == null || limit.isPartial());
```

Comment:
Could you add a test?

Why lack of `limit` context implies partial (rather than final?)

Reply:
Basically:
- If both the context and limit node are partial: new context is partial
- If either the context or the limit node are final: new context is final
- If there is no existing context, new context is either partial or final to match the current node.